src/test/java/com/ncc/aif/ExamplesAndValidationTest.java

```diff
@@ -1112,14 +1119,216 @@ void missingRdfTypeOnNamedNode() {
+                markJustification(eventEdge, compound);
+
+                assertTrue(nistSeedlingValidator.validateKB(model));
+            }
+        }
+
+        @Nested
+        class PreventShotVideo {
+            @Test
+            void invalid() {
+                markShotVideoJustification(model, entity, "source1", "shotId", system, 1d);
+                assertFalse(nistSeedlingValidator.validateKB(model));
+            }
+            @Test
+            void valid() {
+                markKeyFrameVideoJustification(model, entity, "source1", "keyframe",
+                        new BoundingBox(new Point(0, 0), new Point(100, 100)), system, 1d);
+                assertTrue(nistSeedlingValidator.validateKB(model));
+            }
+        }
+
+        @Nested
+        class FlatClusters {
+            @Test
+            void invalid() {
+                markAsPossibleClusterMember(model, eventCluster, entityCluster, .5, system);
+                assertFalse(nistSeedlingValidator.validateKB(model));
+            }
+            @Test
+            void valid() {
+                final Resource newEntity = makeEntity(model, getEntityUri(), system);
+                addType(newEntity, SeedlingOntology.Person);
+                markAsPossibleClusterMember(model, newEntity, entityCluster, .75, system);
+                assertTrue(nistSeedlingValidator.validateKB(model));
+            }
+        }
+
+        @Nested
+        class EverythingClustered {
+            @Test
+            void invalid() {
+                // Test entity, relation, and event. Correct other than being clustered
+                addType(makeEntity(model, getEntityUri(), system), SeedlingOntology.Weapon);
+                addType(makeRelation(model, getUri("relationX"), system),
+                        SeedlingOntology.GeneralAffiliation_APORA);
+                addType(makeEvent(model, getUri("eventX"), system),
+                        SeedlingOntology.Life_BeBorn);
+                assertFalse(nistSeedlingValidator.validateKB(model));
+            }
+            @Test
+            void valid() {
+                final Resource newEntity = makeEntity(model, getEntityUri(), system);
+                addType(newEntity, SeedlingOntology.Weapon);
+                makeClusterWithPrototype(model, getClusterUri(), newEntity, system);
+
+                final Resource relation = makeRelation(model, getUri("relationX"), system);
+                addType(relation, SeedlingOntology.GeneralAffiliation_APORA);
+                makeClusterWithPrototype(model, getClusterUri(), relation, system);
+
+                final Resource newEvent = makeEvent(model, getUri("eventX"), system);
+                addType(newEvent, SeedlingOntology.Life_BeBorn);
+                makeClusterWithPrototype(model, getClusterUri(), newEvent, system);
+
+                assertTrue(nistSeedlingValidator.validateKB(model));
+            }
+        }
+    }
+
```

Comment:
I'm not inclined to. It's something we do currently support, but I'm not sure how useful it is to cluster things with inconsistent types. In other words, I'm not sure it's something we should advertise/promote. Do you think the researchers would find value in this?

Reply:
Well, NIST saw fit to mention it specifically to the aida-data-eval mailing list-- I can only speculate that they thought it was something the researches would want to know-- maybe this is only supported for M18 evaluation but will be removed down the line (because as you suggest, it doesn't sound optimal to cluster things with inconsistent types).  I don't know that it's Next Century's role to decide what should be advertised or promoted.

Maybe the best solution is NOT to add a ValidExample for it, but to add it to the Example AIF that pass and fail NIST validations (AIDA-545) as this requirement (or lack thereof) was discussed in the context of evaluation.

Reply:
One thing which makes this complicated is that you could have multiple type assertions for the same entity which are subject to a `MutualExclusionConstraint`, which would look inconsistent to any code which did not check for that case.

It is also complicated because we might be able to type some nodes at different depths in the ontology hierarchy.